### PR TITLE
refactor(storage_impl): split payment attempt models to domain + diesel

### PR DESCRIPTION
**Pull request by lsampras**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- Creating mirror structs for payment attempt &mandate in data_models
- move needed kv util functions to storage_impl

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by ThisIsMani — DISMISSED** [review]

File: crates/router/src/connector/nuvei/transformers.rs [hunk 1]
```diff
@@ -790,20 +791,28 @@ fn get_card_info<F>(
...
                             details.ok_or(errors::ConnectorError::MissingRequiredField {
                                 field_name: "mandate_data.mandate_type.multi_use",
                             })?
                         }
                     };
-                    let mandate_meta: NuveiMandateMeta =
-                        utils::to_connector_meta_from_secret(Some(details.get_metadata()?))?;
+                    let mandate_meta: NuveiMandateMeta = utils::to_connector_meta_from_secret(
+                        Some(details.get_metadata().ok_or_else(utils::missing_field_err(
+                            "mandate_data.mandate_type.{multi_use|single_use}.metadata",
```
Comment (SanchithHegde) [inline crates/router/src/connector/nuvei/transformers.rs:803]: I'm not sure if we should be returning such error messages which could possibly lead to _some_ confusion.

File: crates/router/src/core/payments.rs [hunk 2]
```diff
@@ -1280,16 +1283,16 @@ pub async fn list_payments(
                 .await
             {
                 Ok(pa) => Some(Ok((pi, pa))),
-                Err(e) => {
-                    if e.current_context().is_db_not_found() {
+                Err(error) => {
+                    if matches!(error.current_context(), StorageError::ValueNotFound(_)) {
```
Comment (SanchithHegde) [inline crates/router/src/core/payments.rs:1287]: What about `StorageError::DatabaseError(Report<DatabaseError::NotFound>)`? As far as I remember, the `is_db_not_found()` function used to handle that as well.

**Review by SanchithHegde — DISMISSED** [review]
Other than that, looks good to me!

File: crates/router/src/core/payments.rs (hunk 2 above)
Comment (lsampras) [reply crates/router/src/core/payments.rs:1284]: this is the StorageError defined in data_models it does not wrap around the database error type

File: crates/router/src/connector/nuvei/transformers.rs (hunk 1 above)
Comment (lsampras) [reply crates/router/src/connector/nuvei/transformers.rs:803]: yeah,
what should we replace this with?

I'm not very aware of the implications of changing this...

File: postman/collection-dir/payme/.event.json
Comment (SanchithHegde) [inline postman/collection-dir/payme/.event.json:1]: I think this file is not on `main` branch?

File: postman/collection-dir/paypal/.event.json
Comment (SanchithHegde) [inline postman/collection-dir/paypal/.event.json:1]: Same with this?

**Review by ThisIsMani — DISMISSED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by harsh-sharma-juspay — DISMISSED** [review]

**Review by SanchithHegde — DISMISSED** [review]

File: crates/router/src/core/payments/helpers.rs [hunk 3]
```diff
@@ -285,16 +288,53 @@ pub async fn get_token_pm_type_mandate_details(
...
     Option<payments::RecurringMandatePaymentData>,
     Option<String>,
 )> {
+    let mandate_data = request.mandate_data.clone().map(|d| MandateData {
+        customer_acceptance: d.customer_acceptance.map(|d| {
+            data_models::mandates::CustomerAcceptance {
+                acceptance_type: match d.acceptance_type {
+                    api::AcceptanceType::Online => data_models::mandates::AcceptanceType::Online,
+                    api::AcceptanceType::Offline => data_models::mandates::AcceptanceType::Offline,
+                },
```
Comment (vspecky) [inline crates/router/src/core/payments/helpers.rs:301]: Better to define this in a conversion trait

**Review by vspecky — CHANGES_REQUESTED** [review]

File: crates/router/src/core/payments/helpers.rs (hunk 3 above)
Comment (lsampras) [reply crates/router/src/core/payments/helpers.rs]: Done!!!

**Review by SanchithHegde — APPROVED** [review]

**Review by vspecky — APPROVED** [review]

**Pull request merged**